src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -2123,19 +2073,19 @@ public override DeclarationKind GetDeclarationKind(SyntaxNode declaration)
             return DeclarationKind.None;
         }
 
-        private bool ParentIsFieldDeclaration(SyntaxNode node)
+        private static bool ParentIsFieldDeclaration(SyntaxNode node)
         {
-            return node != null && node.Parent != null && node.Parent.IsKind(SyntaxKind.FieldDeclaration);
+            return node?.Parent.IsKind(SyntaxKind.FieldDeclaration) ?? false;
         }
 
-        private bool ParentIsEventFieldDeclaration(SyntaxNode node)
+        private static bool ParentIsEventFieldDeclaration(SyntaxNode node)
         {
-            return node != null && node.Parent != null && node.Parent.IsKind(SyntaxKind.EventFieldDeclaration);
+            return node?.Parent.IsKind(SyntaxKind.EventFieldDeclaration) ?? false;
         }
 
-        private bool ParentIsLocalDeclarationStatement(SyntaxNode node)
+        private static bool ParentIsLocalDeclarationStatement(SyntaxNode node)
         {
-            return node != null && node.Parent != null && node.Parent.IsKind(SyntaxKind.LocalDeclarationStatement);
+            return node?.Parent.IsKind(SyntaxKind.LocalDeclarationStatement) ?? false;
```

Comment:
node may be non null, but parent may be null.  You can use IsParentKind to deal with that